lib/yahoo-finance.rb

```diff
@@ -117,6 +117,9 @@ def quotes(symbols_array, columns_array = [
       ret = []
       symbols_array.each_slice(SYMBOLS_PER_REQUEST) do |symbols|
         read_quotes(symbols.join("+"), columns_array).map do |row|
+          if options[:na_as_nil]
+            row.each { |item| item[-1] = nil if item[-1] == 'N/A' }
```

Comment:
The `columns_array` is an arbitrary array, so it's not garanteed that name will be the last element:

```
> YahooFinance::Client.new.quotes(["BVSP", "NATU3.SA", "USDJPY=X"], [:ask, :bid, :last_trade_date])
 => [#<OpenStruct ask="N/A", bid="N/A", last_trade_date="N/A">, #<OpenStruct ask="N/A", bid="N/A", last_trade_date="10/13/2015">, #<OpenStruct ask="119.5530", bid="119.5510", last_trade_date="10/14/2015">]
```

You might want to check all rows for N/A, not just the last